File: backend/LogEntry.py
from datetime import timedelta, datetime
from google.appengine.ext import db


class LogEntry(db.Model):
    job = db.ReferenceProperty()
    name = db.StringProperty(required=True)
    event_time = db.DateTimeProperty(required=True, auto_now_add=True)
    detail = db.StringProperty()

    @staticmethod
    def log_event(job, name, detail):
        entry = LogEntry(job=job, name=name, detail=detail)
        entry.put()

    @staticmethod
    def cleanup():
        to_delete = [] 
        for evt in LogEntry.all(keys_only=True).filter('event_time < ', datetime.utcnow() - timedelta(days=365)):
            to_delete.append(evt)
        
        if len(to_delete) > 0:
            db.delete(to_delete)

File: backend-go/LogEntry.go
package main

import "time"

type LogEntry struct {
	Job       uint `gorm:"foreignkey:JobID"`
	Name      string
	EventTime time.Time
	Detail    string
}

func (env *Env) createLogEntry(job uint, name string, detail string) {
	entry := LogEntry{
		Job:       job,
		Name:      name,
		EventTime: time.Now(),
		Detail:    detail,
	}

	env.db.Create(&entry)
}
